apps/api/src/goods-receipts/goods-receipts.service.ts

```diff
@@ -0,0 +1,75 @@
+import { Injectable } from '@nestjs/common';
+import { CreateGoodsReceiptDto } from './dto/create-goods-receipt.dto';
+import { UpdateGoodsReceiptDto } from './dto/update-goods-receipt.dto';
+import { PrismaService } from "../prisma.service";
+import { GoodsReceipt } from '@prisma/client';
+
+@Injectable()
+export class GoodsReceiptsService {
+  constructor(private readonly prisma: PrismaService) {}
+
+  async create(createGoodsReceiptDto: CreateGoodsReceiptDto) {
+    return this.prisma.goodsReceipt.create({
+      data: {
+        purchase_order_id: createGoodsReceiptDto.purchase_order_id,
+        received_at: createGoodsReceiptDto.received_at,
+        line_items: {
+          create: createGoodsReceiptDto.line_items.map(item => ({
+            item_id: item.item_id,
+            quantity_received: item.quantity_received
+          }))
+        },
+      },
+    });
+  }
+
+  async findAll(): Promise<GoodsReceipt[]> {
+    return this.prisma.goodsReceipt.findMany({
+      include: {
+        purchase_order: true,
+        line_items:true
+      },
+    });
+  }
+
+  async findOne(id: number): Promise<GoodsReceipt> {
+    return this.prisma.goodsReceipt.findUnique({
+      where: { id },
+      include: {
+        line_items: true,
+      },
+    });
+  }
+
+  async update(id: number, updateGoodsReceiptDto: UpdateGoodsReceiptDto): Promise<GoodsReceipt> {
```

Comment:
Looks like you are leveraging the models throughout the code as-is. Do you ever use something like a DTO to control the in/out shape of objects on the backend?

Reply:
I see you have in DTOs now that I scrolled down some haha

Reply:
And above doh...I was jumping around